src/utils/functions/functions.ts

```diff
@@ -22,7 +22,11 @@ export const getFunctionsDir = (
     options: OptionValues
   },
   defaultValue?: string,
-) => options.functions || config.dev?.functions || config.functionsDirectory || defaultValue
+): string | undefined =>
+  ('functions' in options && typeof options.functions === 'string' ? options.functions : null) ??
+  config.dev?.functions ??
+  config.functionsDirectory ??
```

Comment:
Isn't this a behavioral change? Won't this now allow `config.dev.functions` and `config.functionsDirectory` to be an empty string? Or do we sanitize empty strings out somewhere?

Reply:
I'm not sure how you could even end up with an empty string, other than explicitly using `""` in your `netlify.toml` or something... 🤔

I feel like it isn't worth the added complexity to this line to support ignoring hypothetical empty strings here, but in the interest of limiting behavioural changes I'll go ahead and add the conditions 🙈.

Reply:
I don't think it's worth the added complexity here or the change risk, however low that risk is (you're still adding in new behavior by checking for strings with only whitespace in them). I don't think that bundling it in with a clear and totally unrelated bug is really the way to go about this change, either.

Reply:
I thought that `trim()` wouldn't be controversial, but alright, I've reverted that too.